tests/phpunit/includes/bootstrap.php

```diff
@@ -46,13 +46,83 @@
-	echo 'Error: You need to run `composer update` before running the tests.' . PHP_EOL;
-	echo 'You can still use a PHPUnit phar to run them, but the dependencies do need to be installed.' . PHP_EOL;
+/*
+ * Load the PHPUnit Polyfills autoloader.
+ *
+ * The PHPUnit Polyfills are a requirement for the WP Test Suite.
+ *
+ * Plugin/theme integration tests can handle this in any of the following ways:
+ * - When using a full WP install: run `composer update` for the WP install prior to running the tests.
+ * - When using a partial WP test suite install:
+ *   - Add a `yoast/phpunit-polyfills:"^1.0.1"` requirement to their own `composer.json` file.
+ *   - And then:
+ *     - Either load the PHPUnit Polyfills autoload file prior to running the WP Core bootstrap file.
+ *     - Or declare a `WP_TESTS_PHPUNIT_POLYFILLS_PATH` constant pointing to the root directory
+ *       of the PHPUnit Polyfills installation.
+ *       This constant can be declared in the `phpunit.xml[.dist]` file like so:
+ *       `<php><const name="WP_TESTS_PHPUNIT_POLYFILLS_PATH" value="path/to/yoast/phpunit-polyfills"/></php>
+ *       or can be declared as a PHP constant in the `wp-tests-config.php` file or the plugin/theme
+ *       test bootstrap file.
+ */
+if ( ! class_exists( 'Yoast\PHPUnitPolyfills\Autoload' ) ) {
+	// Default location of the autoloader for WP Core test runs.
+	$phpunit_polyfills_autoloader = __DIR__ . '/../../../vendor/yoast/phpunit-polyfills/phpunitpolyfills-autoload.php';
+	$phpunit_polyfills_error      = false;
+
+	// Allow for a custom installation location to be provided for plugin/theme integration tests.
+	if ( defined( 'WP_TESTS_PHPUNIT_POLYFILLS_PATH' ) ) {
+		$phpunit_polyfills_path = WP_TESTS_PHPUNIT_POLYFILLS_PATH;
+
+		if ( is_string( WP_TESTS_PHPUNIT_POLYFILLS_PATH )
+			&& '' !== WP_TESTS_PHPUNIT_POLYFILLS_PATH
+		) {
+			$phpunit_polyfills_path = rtrim( $phpunit_polyfills_path, '/\\' );
+			$phpunit_polyfills_path = realpath( $phpunit_polyfills_path . '/phpunitpolyfills-autoload.php' );
+			if ( false !== $phpunit_polyfills_path ) {
+				$phpunit_polyfills_autoloader = $phpunit_polyfills_path;
+			} else {
+				$phpunit_polyfills_error = true;
+			}
+		} else {
+			$phpunit_polyfills_error = true;
+		}
+	}
+
+	if ( $phpunit_polyfills_error || ! file_exists( $phpunit_polyfills_autoloader ) ) {
+		echo 'Error: The PHPUnit Polyfills library is a requirement for running the WP test suite.' . PHP_EOL;
+		if ( isset( $phpunit_polyfills_path ) ) {
+			printf(
+				'The PHPUnit Polyfills autoload file was not found in "%s"' . PHP_EOL,
+				WP_TESTS_PHPUNIT_POLYFILLS_PATH
+			);
+			echo 'Please verify that the file path provided in the WP_TESTS_PHPUNIT_POLYFILLS_PATH constant is correct.' . PHP_EOL;
+		} else {
+			echo 'You need to run `composer update` before running the tests.' . PHP_EOL;
+			echo 'You can still use a PHPUnit phar to run them, but the dependencies do need to be installed.' . PHP_EOL;
```

Comment:
When I first read the error message, I must admit that I was confused. Took me a moment to realize why I needed to run `composer update` in the case where I wanted to use the phar. I wonder if the message could be more clear to avoid any confusion.

```suggestion
			echo 'You can still use a PHPUnit phar to run them, but the dependencies do need to be installed, which is done when running `composer update`.' . PHP_EOL;
```

Reply:
IMO that is redundant as that is exactly what the first line tells the user to do. This is not the place for a "how to use Composer and how  does it work" tutorial.

Reply:
It's not.  And I do see your point. 

BUT there's a disconnect between the first sentence of why you need to run `composer update` and the second sentence of using phar. It could cause confusion for extenders.

That said, we could go with what you have and then iterate if there is confusion.

Reply:
@hellofromtonya I agree there is a disconnect. I've updated the message text to try and make things clearer. Could you please have another look ?

Reply:
Oh the new message is much much better! It's clear what the extender needs to do and why👍